pandas/core/groupby/groupby.py

```diff
@@ -1615,8 +1612,8 @@ def _wrap_applied_output(
 
     @final
     def _numba_prep(self, data: DataFrame):
-        ids, _, ngroups = self.grouper.group_info
-        sorted_index = self.grouper._sort_idx
```

Comment:
off topic: should we privatize grouper->_grouper?

Reply:
I'd be +1 on this; I can get this deprecation into 2.2.